GibbsSampling.Library/CompositeVector/CompositeVectorFunctions.fs

```diff
@@ -31,7 +32,7 @@ module Functions =
     /// Create FrequencyCompositeVector based on BioArray and excludes the specified segment.
     let createFCVWithout (motiveLength:int) (position:int) (resSource:BioArray.BioArray<#IBioItem>) =
         let backGroundCounts = new FrequencyCompositeVector()
-        Array.append resSource.[0..(position-1)] resSource.[(position+motiveLength)..]
+        mergeArrays resSource.[0..(position-1)] resSource.[(position+motiveLength)..]
```

Comment:
what is this meant to do? append, just joins two arrays, right?

Reply:
Yes but we exclude the motif, based on its start position and length ->
We split the array arround the motif (position + length) and merge it again.

Reply:
This?

````fsharp
let array1 = [|0..100|]

// mapi approach
let test (motiveStart: int) (motiveLength: int) (arr2: 'b array) =
    arr2
    |> Array.mapi (fun i (x: 'b) -> 
      let motiveEnd = motiveStart+motiveLength
      if i >= motiveStart && i < motiveEnd then
        None
      else
        Some x
    )
    |> Array.choose id

// fold approach
let test2 (motiveStart: int) (motiveLength: int) (arr2: 'b array) =
    arr2
    |> Array.fold (fun (acc: 'b list, i: int) (x: 'b) -> 
      let motiveEnd = motiveStart+motiveLength
      if i >= motiveStart && i < motiveEnd then
        (acc, i+1)
      else
        (x::acc, i+1)
    ) ([],0)
    |> fst
    |> List.rev
    |> List.toArray

```

not sure which is more optimized under the hood.